packages/dcos-integration-test/extra/resiliency/test_aws_cf.py

```diff
@@ -1,52 +1,74 @@
+import logging
 import os
-from functools import partial
 
 import pytest
+import retrying
 
 import test_util.aws
 import test_util.helpers
 from test_util.helpers import retry_boto_rate_limits
 
+log = logging.getLogger(__name__)
+
 ENV_FLAG = 'ENABLE_RESILIENCY_TESTING'
 
-pytestmark = pytest.mark.skipif(
+skip_if_resiliency_not_enabled = pytest.mark.skipif(
     ENV_FLAG not in os.environ or os.environ[ENV_FLAG] != 'true',
     reason='Must explicitly enable resiliency testing with {}'.format(ENV_FLAG))
 
+skip_if_not_aws_stack = pytest.mark.skipif(
+    'AWS_STACK_NAME' not in os.environ,
+    reason='Must use an AWS Cloudformation run this test!')
+
+pytestmark = [skip_if_resiliency_not_enabled, skip_if_not_aws_stack]
+
 
 @pytest.fixture(scope='session')
-def dcos_launchpad(dcos_api_session):
-    """Interface for direct integration to dcos_launchpad hardware
-    Currently only supports AWS CF with AWS VPC coming soon
-    """
-    if 'AWS_STACK_NAME' not in os.environ:
-        # TODO(mellenburg): update when advanced templates are merged
-        pytest.skip('Must use a AWS Cloudformation to run test')
-    stack_name = os.environ['AWS_STACK_NAME']
+def boto_wrapper(dcos_api_session):
     aws_region = os.environ['AWS_REGION']
     aws_access_key_id = os.environ['AWS_ACCESS_KEY_ID']
     aws_secret_access_key = os.environ['AWS_SECRET_ACCESS_KEY']
-    bw = test_util.aws.BotoWrapper(aws_region, aws_access_key_id, aws_secret_access_key)
-    return test_util.aws.DcosCfSimple(stack_name, bw)
+    return test_util.aws.BotoWrapper(aws_region, aws_access_key_id, aws_secret_access_key)
+
+
+@pytest.fixture(scope='session')
+def dcos_stack(boto_wrapper):
+    """ Works with either Zen or Simple cloud formations
+    """
+    stack = test_util.aws.fetch_stack(os.environ['AWS_STACK_NAME'], boto_wrapper)
+    if isinstance(stack, test_util.aws.VpcCfStack):
+        pytest.skip('Onprem Vpc not currently supported')
+    return stack
 
 
 @pytest.mark.last
-def test_agent_failure(dcos_launchpad, dcos_api_session, vip_apps):
-    # make sure the app works before starting
+def test_agent_failure(dcos_stack, boto_wrapper, dcos_api_session, vip_apps):
+    # Accessing AWS Resource objects will trigger a client describe call.
+    # As such, any method that touches AWS APIs must be wrapped to avoid
+    # CI collapse when rate limits are inevitably reached
+    @retry_boto_rate_limits
```

Comment:
Also, unless I misread my implementation, I shouldnt be having that issue with yield. Rather, it will retry to walk all the way through the iterator without encountering an error.